src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDB = userService.findByEmail(email);
+        if (userFromDB.isEmpty()
+                || userFromDB.get().getPassword()
+                .equals(HashUtil.hashPassword(password, userFromDB.get().getSalt()))) {
+            throw new AuthenticationException("Can't authenticate user");
+        }
+        return userFromDB.get();
```

Comment:
you use the get() method three times. Perhaps it is better to make the type User instead of Option User>? This way you will get rid of the isEmpty check